Makefile

```diff
@@ -77,21 +77,71 @@ BUILD = _build/build-$(BOARD)
 # Board specific
 -include src/boards/$(BOARD)/board.mk
 
-# MCU_SUB_VARIANT can be nrf52 (nrf52832), nrf52833, nrf52840
-ifeq ($(MCU_SUB_VARIANT),nrf52)
-  SD_NAME = s132
-  DFU_DEV_REV = 0xADAF
-  CFLAGS += -DNRF52 -DNRF52832_XXAA -DS132
-else ifeq ($(MCU_SUB_VARIANT),nrf52833)
-  SD_NAME = s140
+#---------------------------------
+# Select the softdevice to build for
+#---------------------------------
+ifeq ($(SD),s140)
+  CFLAGS += -DS140 -DBLE
   DFU_DEV_REV = 52840
-  CFLAGS += -DNRF52833_XXAA -DS140
-else ifeq ($(MCU_SUB_VARIANT),nrf52840)
-  SD_NAME = s140
+  ifeq ($(MCU_SUB_VARIANT),nrf52833)
+    CFLAGS += -DNRF52833_XXAA
+  else ifeq ($(MCU_SUB_VARIANT),nrf52840)
+    CFLAGS += -DNRF52840_XXAA
+  else
+    $(error Sub Variant $(MCU_SUB_VARIANT) is invalid for softdevice $(SD))
+  endif
+else ifeq ($(SD),s340)
+  $(info Build specified supporting both ANT and BLE)
+  $(warning Ensure you download a dualstack softdevice from thisisant.com)
+  $(warning Make sure to compile against the dualstack softdevice headers)
+  CFLAGS += -DS340 -DBOTH
   DFU_DEV_REV = 52840
-  CFLAGS += -DNRF52840_XXAA -DS140
+  ifeq ($(MCU_SUB_VARIANT),nrf52833)
+    CFLAGS += -DNRF52833_XXAA
+  else ifeq ($(MCU_SUB_VARIANT),nrf52840)
+    CFLAGS += -DNRF52840_XXAA
+  else
+    $(error Sub Variant $(MCU_SUB_VARIANT) is invalid for softdevice $(SD))
+  endif
+else ifeq ($(SD),s132)
+  CFLAGS += -DS132 -DBLE
+  DFU_DEV_REV = 0xADAF
+  ifeq ($(MCU_SUB_VARIANT),nrf52)
+    CFLAGS += -DNRF52832_XXAA
+  else
+    $(error Sub Variant $(MCU_SUB_VARIANT) is invalid for softdevice $(SD))
+  endif
+else ifeq ($SD, s332)
+  $(info Build specified supporting both ANT and BLE)
+  $(warning Ensure you download a dualstack softdevice from thisisant.com)
+  $(warning Make sure to compile against the dualstack softdevice headers)
+  CFLAGS += -DS332 -DBOTH
+  DFU_DEV_REV = 0xADAF
+  ifeq ($(MCU_SUB_VARIANT),nrf52)
+    CFLAGS += -DNRF52832_XXAA
+  else
+    $(error Sub Variant $(MCU_SUB_VARIANT) is invalid for softdevice $(SD))
+  endif
+else ifeq ($SD,)

```

Comment:
Too many duplication here, this should be checked before the whole if, and SD should get the default only to S132/S140 only. Then the ifeq() sequence take care of all the CFLAGS.